recipes/default.rb

```diff
@@ -63,28 +68,74 @@
   action [:start, :enable]
 end
 
-user "jenkins" do
+user agent_username  do
   shell "/bin/bash"
   manage_home true
 end
-sudo "jenkins" do
-  user "jenkins"
+sudo agent_username do
+  user agent_username
   nopasswd true
 end
 
 # Add agent user to the docker group to allow them to build and run docker
 # containers.
 group 'docker' do
   append true
-  members 'jenkins'
+  members agent_username
   action :manage # Group should be created by docker package.
 end
 
-directory "/home/jenkins/jenkins-agent"
-agent_jar_url = node["osrf_jenkins_agent"]["agent_jar_url"]
-if agent_jar_url.nil? || agent_jar_url.empty?
-  agent_jar_url = "#{node["osrf_jenkins_agent"]["jenkins_url"]}/jnlpJars/agent.jar"
+
+# TODO: how to read attributes from chef-osrf plugins into this cookbook
+# swarm_client_version = node['jenkins-plugins']['swarm']
```

Comment:
I think that the main difference is that `cookbook-ros-buildfarm` hosts agent/server/repo while at this point the `osrf_jenkins_agent` is no more than the agent. Having a `plugins.rb`file in this repo has little sense since they are not consumed by the agent. Not sure the best way of fixing this.

Reply:
That's a good point. Long term, I could see integrating the agent recipe into the same cookbook as the server and packages hosts but for the time being I think defining a default attribute in the cookbook and then making sure that the default attributes match between the two cookbooks in the deployment environment.